src/ReactiveUI.Blazor/Registrations.cs

```diff
@@ -27,18 +25,13 @@ public void Register(Action<Func<object>, Type> registerFunction)
 
             registerFunction(() => new PlatformOperations(), typeof(IPlatformOperations));
 
-#if NETSTANDARD
-            if (WasmPlatformEnlightenmentProvider.IsWasm)
+            if (Type.GetType("Mono.Runtime") != null)
             {
-                RxApp.TaskpoolScheduler = WasmScheduler.Default;
-                RxApp.MainThreadScheduler = WasmScheduler.Default;
-            }
-            else
-#endif
-            {
-                RxApp.TaskpoolScheduler = TaskPoolScheduler.Default;
-                RxApp.MainThreadScheduler = new WaitForDispatcherScheduler(() => CurrentThreadScheduler.Instance);
+                PlatformEnlightenmentProvider.Current.EnableWasm();
             }
+
+            RxApp.TaskpoolScheduler = TaskPoolScheduler.Default;
+            RxApp.MainThreadScheduler = new WaitForDispatcherScheduler(() => CurrentThreadScheduler.Instance);
```

Comment:
We should never have to wait for the dispatcher to get CurrentThread, there is no dispatcher!

Reply:
I tried to change as little as possible but I can remove that line and test again if you like.

Reply:
The name of `WaitForDispatcherScheduler` is deceptive. It just checks for an exception, if there is one just returns the `CurrentThreadScheduler` instead.

Reply:
So it has no reliance on the dispatcher directly. It was intended to be used with the DispatcherScheduler though.